Extensions of LineInformation, StopInformation and TripInformation St…

ThomasKling opened the pull request:
…ructure plus Number of Error Messages in DMS reduced from 10 to 0.

ThomasKling committed:
Extensions of LineInformation, StopInformation and TripInformation Structure plus Number of Error Messages in DMS reduced from 10 to 0.

ThomasKling committed:
Update IBIS-IP_common_V2.4.xsd

ThomasKling commented:
LineSymbolText redefined to InternationTextType

dirkweisser commented:
In the document which went through the review process, we deleted the optional LineCode of type IBIS-IP.int, introduced an optional LineSymbolCode of type IBIS-IP.int and had the LinePublicCode of this PullRequest before the new LineSymbolCode. We either had to change the document or the xsd.

dirkweisser commented:
In the document we had in the TripInformationStructure the new BlockNumber before the PatternNumber. We either had to change the document or the xsd.

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="FareZone" type="IBIS-IP.NMTOKEN" minOccurs="0" maxOccurs="unbounded"/>
+			<xs:element name="StopShortName" type="InternationalTextType" minOccurs="0" maxOccurs="unbounded"/>
```
```suggestion
<xs:element name="StopShortName" type="InternationalTextType" minOccurs="0" maxOccurs="unbounded"/>
<xs:annotation>
<xs:documentation>Short name of the stop in the planning program and ITCS. To identify a stop on media with limited display length, e.g. older interior displays, on-board computer or ticket printer terminal.
</xs:documentation>
```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+					<xs:documentation>Information on Connections or Interchanges; not in StopSequence because of RealTimeInformation</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="FareZone" type="IBIS-IP.NMTOKEN" minOccurs="0" maxOccurs="unbounded"/>
```
```suggestion
			<xs:element name="FareZone" type="IBIS-IP.NMTOKEN" minOccurs="0" maxOccurs="unbounded"/>
			<xs:annotation>
<xs:documentation>Valid fare zone at this stop point</xs:documentation>
</xs:annotation>
```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+			</xs:element>
+			<xs:element name="FareZone" type="IBIS-IP.NMTOKEN" minOccurs="0" maxOccurs="unbounded"/>
+			<xs:element name="StopShortName" type="InternationalTextType" minOccurs="0" maxOccurs="unbounded"/>
+			<xs:element name="StopLongNo" type="IBIS-IP.int" minOccurs="0" maxOccurs="1"/>			
```
```suggestion
			<xs:element name="StopLongNo" type="IBIS-IP.int" minOccurs="0" maxOccurs="1"/>
			<xs:annotation>
					<xs:documentation>Stop number that is used throughout the transport association (PTA) for activation of ticket printers and ticket validators. This number is not necessarily unique and is only used for stops that are used productively. This number is not required for non-productive stops (e.g. depots, parking facilities, car parks).
</xs:documentation>
				</xs:annotation>			
			
```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+			<xs:element name="FareZone" type="IBIS-IP.NMTOKEN" minOccurs="0" maxOccurs="unbounded"/>
+			<xs:element name="StopShortName" type="InternationalTextType" minOccurs="0" maxOccurs="unbounded"/>
+			<xs:element name="StopLongNo" type="IBIS-IP.int" minOccurs="0" maxOccurs="1"/>			
+			<xs:element name="PointNumber" type="IBIS-IP.int" minOccurs="0" maxOccurs="1"/>			
```
```suggestion
			<xs:element name="PointNumber" type="IBIS-IP.int" minOccurs="0" maxOccurs="1"/>			
			<xs:annotation>
					<xs:documentation>Transport company-wide internal unique number of a stopping point in the network.
</xs:documentation>
				</xs:annotation>
```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+			<xs:element name="StopShortName" type="InternationalTextType" minOccurs="0" maxOccurs="unbounded"/>
+			<xs:element name="StopLongNo" type="IBIS-IP.int" minOccurs="0" maxOccurs="1"/>			
+			<xs:element name="PointNumber" type="IBIS-IP.int" minOccurs="0" maxOccurs="1"/>			
+			<xs:element name="StopGlobalID" type="IBIS-IP.string" minOccurs="0" maxOccurs="1"/>			
```
```suggestion
			<xs:element name="StopGlobalID" type="IBIS-IP.string" minOccurs="0" maxOccurs="1"/>	
			<xs:annotation>
					<xs:documentation>International stop ID e.g. in Germany according to VDV432 (DHID), internationally according to NeTEx, NAPTAN.
					</xs:documentation>
				</xs:annotation>		
```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+			<xs:element name="StopLongNo" type="IBIS-IP.int" minOccurs="0" maxOccurs="1"/>			
+			<xs:element name="PointNumber" type="IBIS-IP.int" minOccurs="0" maxOccurs="1"/>			
+			<xs:element name="StopGlobalID" type="IBIS-IP.string" minOccurs="0" maxOccurs="1"/>			
+			<xs:element name="StopPointGlobalID" type="IBIS-IP.string" minOccurs="0" maxOccurs="1"/>						
```
```suggestion
			<xs:element name="StopPointGlobalID" type="IBIS-IP.string" minOccurs="0" maxOccurs="1"/>	
			<xs:annotation>
					<xs:documentation>International stopping point ID e.g. in Germany according to VDV432 (DHID), internationally according to NeTEx, NAPTAN.
					</xs:documentation>
				</xs:annotation>					
```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+					<xs:documentation>The run number (Kurs-Nummer) the trip is operated </xs:documentation>
+				</xs:annotation>
+			</xs:element>						
+			<xs:element name="PatternNumber" type="IBIS-IP.int" minOccurs="0">
```
```suggestion
			<xs:element name="BlockNumber" type="IBIS-IP.int"  minOccurs="0" maxOccurs="1"/>	
				<xs:annotation>
					<xs:documentation>Unique block number per day of operational day. Used for logon on the vehicle in block operation.
					</xs:documentation>
				</xs:annotation>
			<xs:element name="PatternNumber" type="IBIS-IP.int" minOccurs="0">
```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+					<xs:documentation>The path destination number (RoutenZiel-Nummer) the trip is operated </xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="BlockNumber" type="IBIS-IP.int"  minOccurs="0" maxOccurs="1"/>						
```
```suggestion
			<xs:element name="BlockNumber" type="IBIS-IP.int"  minOccurs="0" maxOccurs="1"/>	
				<xs:annotation>
					<xs:documentation>Unique block number per day of operational day. Used for logon on the vehicle in block operation.
					</xs:documentation>
				</xs:annotation>			
```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="BlockNumber" type="IBIS-IP.int"  minOccurs="0" maxOccurs="1"/>						
+			<xs:element name="ExternalVehicleJourneyRef" type="IBIS-IP.string"  minOccurs="0" maxOccurs="1"/>			
```
```suggestion
			<xs:element name="ExternalVehicleJourneyRef" type="IBIS-IP.string"  minOccurs="0" maxOccurs="1"/>	
			<xs:annotation>
					<xs:documentation>
			International trip ID e.g. in Germany according to VDV433 (DFID), internationally according to NeTEx.
</xs:documentation>
				</xs:annotation>									
```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+			<xs:element name="LineName" type="InternationalTextType" minOccurs="0" maxOccurs="unbounded"/>
+			<xs:element name="LineShortName" type="InternationalTextType" minOccurs="0" maxOccurs="unbounded"/>
+			<xs:element name="LineNumber" type="IBIS-IP.int" minOccurs="0"/>
+			<xs:element name="LineCode" type="IBIS-IP.int" minOccurs="0">
```
```suggestion
			<xs:element name="LineCode" type="IBIS-IP.int" minOccurs="0">
				<xs:annotation>
					<xs:documentation>Unique additional number for displaying symbols or special lines on and in the vehicle. This additional number is used as a supplement to the LineNumber. For example, for the display of symbols such as airport, P+R, school bus or the coding of special lines such as night buses, rail replacement services, substitute lines.
					</xs:documentation>
				</xs:annotation>
```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+			<xs:element name="LineShortName" type="InternationalTextType" minOccurs="0" maxOccurs="unbounded"/>
+			<xs:element name="LineNumber" type="IBIS-IP.int" minOccurs="0"/>
+			<xs:element name="LineCode" type="IBIS-IP.int" minOccurs="0">
+				<xs:annotation>
```
```suggestion
```

dirkweisser replied:
```suggestion

```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+			<xs:element name="LineNumber" type="IBIS-IP.int" minOccurs="0"/>
+			<xs:element name="LineCode" type="IBIS-IP.int" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>identifier of a symbol number , which is used by displays see LINIEN_CODE in VDV 452 chapter 9.7.2 REC_LID</xs:documentation>
```
```suggestion
```

dirkweisser replied:
```suggestion

```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+			<xs:element name="LineCode" type="IBIS-IP.int" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>identifier of a symbol number , which is used by displays see LINIEN_CODE in VDV 452 chapter 9.7.2 REC_LID</xs:documentation>
+				</xs:annotation>
```
```suggestion
```

dirkweisser replied:
```suggestion

```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+					<xs:documentation>identifier of a symbol number , which is used by displays see LINIEN_CODE in VDV 452 chapter 9.7.2 REC_LID</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="LinePublicCode" type="IBIS-IP.string" minOccurs="0" maxOccurs="1"/>			
```
```suggestion		
			<xs:element name="LinePublicCode" type="IBIS-IP.string" minOccurs="0" maxOccurs="1"/>
			<xs:annotation>
					<xs:documentation> Published line identification for the passenger. For displaying the line on all media such as line network maps, timetable printouts, passenger information on the internet, on signs, the stop displays, on the exterior and interior displays of the vehicles. 
					</xs:documentation>
				</xs:annotation>
```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="LinePublicCode" type="IBIS-IP.string" minOccurs="0" maxOccurs="1"/>			
+			<xs:element name="LineSymbolText" type="InternationalTextType" minOccurs="0" maxOccurs="unbounded"/>			
```
```suggestion
			<xs:element name="LineSymbolText" type="InternationalTextType" minOccurs="0" maxOccurs="unbounded"/>
			<xs:annotation>
					<xs:documentation>Specification of a xml formatted text to reference an symbol for special lines. Used as alternative to the LineSymbolCode. For example for displaying an airport symbol the following string could be used:  <icon type=" airport "> airport </icon>
					</xs:documentation>
				</xs:annotation>
```

dirkweisser commented on IBIS-IP_common_V2.4.xsd:
```diff
@@ -0,0 +1,1061 @@
+			</xs:element>
+			<xs:element name="LinePublicCode" type="IBIS-IP.string" minOccurs="0" maxOccurs="1"/>			
+			<xs:element name="LineSymbolText" type="InternationalTextType" minOccurs="0" maxOccurs="unbounded"/>			
+			<xs:element name="ExternalLineRef" type="IBIS-IP.string" minOccurs="0" maxOccurs="1"/>			
```
```suggestion
			<xs:element name="ExternalLineRef" type="IBIS-IP.string" minOccurs="0" maxOccurs="1"/>	
			<xs:annotation>
					<xs:documentation>International line ID e.g. in Germany according to VDV433 DLID, internationally according to NeTEx.</xs:documentation>
				</xs:annotation>		
```

ThomasKling commented:
In the xsd the LineCode is available.
In a WebCo we have agreed to reuse LineCode and not to rename it.
Therefore the document has to be changed.

Best regards
Thomas

Von: Dirk Weißer ***@***.***>
Gesendet: Mittwoch, 4. Januar 2023 11:44
An: VDVde/VDV301 ***@***.***>
Cc: Thomas Kling ***@***.***>; State change ***@***.***>
Betreff: Re: [VDVde/VDV301] Extensions of LineInformation, StopInformation and TripInformation St… (PR #25)


EXTERNAL: Do not click links or open attachments if you do not recognize the sender.

In the document which went through the review process, we deleted the optional LineCode of type IBIS-IP.int, introduced an optional LineSymbolCode of type IBIS-IP.int and had the LinePublicCode of this PullRequest before the new LineSymbolCode. We either had to change the document or the xsd.

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/VDVde/VDV301/pull/25*issuecomment-1370763048__;Iw!!I_DbfM1H!EJwIbRTLfsHYTWHzDJyJ_18dt_GRgomVy0UgQQ9rcNwUNICJKSkAFzQiDvTIQKzQx-BviBIJmi3-Zwa6NcBgqfMoNpyyUyi9$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/ANALFIT6TSBNZJ47ZDW24WDWQVIALANCNFSM6AAAAAAQZS2HDE__;!!I_DbfM1H!EJwIbRTLfsHYTWHzDJyJ_18dt_GRgomVy0UgQQ9rcNwUNICJKSkAFzQiDvTIQKzQx-BviBIJmi3-Zwa6NcBgqfMoNsquCrYv$>.
You are receiving this because you modified the open/close state.Message ID: ***@***.******@***.***>>

dirkweisser commented:
ok - changed in my proposal. 

What about the order? First LineCode or first LinePublicCode?
